### ISSUE TITLE
model architectures and pretrained models to support

### ISSUE THREAD

**Issue by dmarx** [dmarx/Multi-Modal-Comparators#2]
- [ ]  CLIP 
  - [ ]  [OpenAI/CLIP](https://github.com/openai/CLIP)
  - [ ] [mlfoundations/open_clip](https://github.com/mlfoundations/open_clip)
  - [ ] [sberbank-ai/ru-clip](https://github.com/sberbank-ai/ru-clip)
  - [ ] https://github.com/Sense-GVT/DeCLIP
- [ ]  CLOOB
  - [crowsonkb/cloob-training](https://github.com/crowsonkb/cloob-training)
  - pretrained model from CLOOB authors: https://github.com/ml-jku/cloob/blob/master/src/notebooks/zeroshot.ipynb
- [ ]  SLIP 
  - [ ] [facebookresearch/SLIP](https://github.com/facebookresearch/SLIP)
  - [ ] https://github.com/Sense-GVT/DeCLIP
- [ ]  BLIP - [salesforce/BLIP](https://github.com/salesforce/BLIP)
- [ ]  CLIP-JAX
- [ ] AudioCLIP 
  - https://github.com/AndreyGuzhov/AudioCLIP
  - https://github.com/NotNANtoN/AudioCLIP
- [ ] DeCLIP
  - [ ] https://github.com/Sense-GVT/DeCLIP
- [ ] FILIP
  - [ ] https://github.com/Sense-GVT/DeCLIP
- [ ] DeFILIP
  - [ ] https://github.com/Sense-GVT/DeCLIP

References for more variants:

https://github.com/lucidrains/x-clip
https://github.com/Sense-GVT/DeCLIP

Potentially in scope, lower priority
* https://github.com/j-min/VL-T5
* https://github.com/ashkamath/mdetr
* https://github.com/sberbank-ai/ru-dolph
* https://github.com/drboog/Lafite
* https://github.com/pbaylies/Augmented_CLIP
* ALIGN (is this even public? or just big?) 
  - https://ai.googleblog.com/2021/05/align-scaling-up-visual-and-vision.html
  - https://arxiv.org/abs/2102.05918
* ALBEF 
  - https://arxiv.org/abs/2102.05918
  - https://github.com/salesforce/ALBEF
* ArtEmis - Affective language for Visual Art
  - https://github.com/optas/artemis
* https://github.com/CompVis/net2net
* LiT (not sure if there are public pre-trained models) - https://arxiv.org/abs/2111.07991
* NUWA (very unlikely they'll release a reptrained model) 
  - https://github.com/microsoft/NUWA
  - https://github.com/lucidrains/nuwa-pytorch

Older stuff
* DAMSM - https://github.com/taoxugit/AttnGAN
* https://github.com/sidward14/Style-AttnGAN
* https://github.com/luogen1996/MCN
* ViLBERT - https://github.com/facebookresearch/vilbert-multi-task

VQA is sort of a generalization of vision language co-training... TBD.
* https://github.com/facebookresearch/Detic
* https://github.com/henghuiding/Vision-Language-Transformer

MAGMA could be another useful approach to promote multi-lingual support

https://github.com/Aleph-Alpha/magma

**Comment by dmarx** [comment]
https://github.com/navervision/KELIP

**Comment by apolinario** [comment]
LiT: they released some models last week https://github.com/google-research/vision_transformer#lit-models

Audio: besides AudioCLIP there's also wav2clip with a different approach: https://github.com/descriptinc/lyrebird-wav2clip 
![image](https://user-images.githubusercontent.com/788417/163956150-40b9e30d-a7b4-48f5-a302-831b4a1b72a9.png)

**Comment by dmarx** [comment]
https://github.com/allenai/reclip

**Comment by dmarx** [comment]
https://github.com/facebookresearch/Detic

**Comment by dmarx** [comment]
https://github.com/sallymmx/ActionCLIP

**Comment by dmarx** [comment]
https://github.com/ChenRocks/UNITER

**Comment by dmarx** [comment]
https://github.com/raoyongming/DenseCLIP

**Comment by dmarx** [comment]
https://github.com/ttlmh/Bridge-Prompt

**Comment by apolinario** [comment]
https://github.com/sonoisa/clip-japanese

**Comment by dmarx** [comment]
* https://mmf.sh/docs/notes/projects
  * older stuff, e.g. ViLBET
* https://github.com/facebookresearch/multimodal

**Comment by Philipuss1** [comment]
KELIP is also a thing: https://github.com/navervision/KELIP/ 😄

**Comment by dmarx** [comment]
SLIP demo install using dapm and loading weights using old strategy from DD: https://github.com/alembics/disco-diffusion/commit/c509aa1b9c00b2323a1fd95c5b0fc667bb12be4c

```
!pip install napm
import napm
url = 'https://github.com/facebookresearch/SLIP'
napm.pseudoinstall_git_repo(url, add_install_dir_to_path=True)

import SLIP
from SLIP.models import SLIP_VITB16, SLIP, SLIP_VITL16

!wget https://dl.fbaipublicfiles.com/slip/slip_base_100ep.pt

import torch
sd = torch.load('slip_base_100ep.pt', map_location=torch.device('cpu') )
real_sd = {}
for k, v in sd['state_dict'].items():
  new_key = '.'.join(k.split('.')[1:]) # strips "module" prefix. sure, why not.
  #print(k, new_key) 
  real_sd[new_key] = v
del sd

SLIPB16model = SLIP_VITB16(ssl_mlp_dim=4096, ssl_emb_dim=256)
SLIPB16model.load_state_dict(real_sd)
```

**Comment by dmarx** [comment]
CLOOB demo using dapm

```
!pip install git+https://github.com/openai/CLIP
import napm

url = "https://github.com/crowsonkb/cloob-training"
napm.pseudoinstall_git_repo(url, package_name='cloob')

import cloob
from cloob.cloob_training import model_pt, pretrained

config = pretrained.get_config('cloob_laion_400m_vit_b_16_16_epochs')
model = model_pt.get_pt_model(config)
checkpoint = pretrained.download_checkpoint(config)
model.load_state_dict(model_pt.get_pt_params(config, checkpoint), )
#model.eval().requires_grad_(False).to('cuda')
```

**Comment by dmarx** [comment]
CoCa https://arxiv.org/abs/2205.01917

**Comment by dmarx** [comment]
OTTER https://github.com/facebookresearch/OTTER

**Comment by dmarx** [comment]
ruDOLPH
* https://colab.research.google.com/drive/1gmTDA13u709OXiAeXWGm7sPixRhEJCga?usp=sharing
* https://github.com/ai-forever/ru-dolph

**Comment by dmarx** [comment]
https://socraticmodels.github.io/

**Comment by dmarx** [comment]
https://github.com/yxuansu/MAGIC

**Comment by apolinario** [comment]
https://github.com/rinnakk/japanese-clip (not the same as [this](https://github.com/dmarx/Multi-Modal-Comparators/issues/2#issuecomment-1104324296))

**Comment by apolinario** [comment]
This is a very large and seemingly very good CLIP in Chinese that @Dango233 has shown me: https://wukong-dataset.github.io/wukong-dataset/benchmark.html

One problem though: It's pre-trained weights are on Mindspore (Huawei's PyTorch) so someone would need to convert that...

**Comment by Dango233** [comment]
> 

https://github.com/mindspore-ai/models/tree/master/research/mm/wukong

**Comment by dmarx** [comment]
maybe just the fine tuned model?

https://github.com/j-min/clip-caption-reward

**Comment by apolinario** [comment]
A new (better, it seems) Multilingual CLIP https://github.com/FreddeFrallan/Multilingual-CLIP

**Comment by rom1504** [comment]
@apolinario indeed and now it's packaged properly on pypi
it's available for testing at https://rom1504.github.io/clip-retrieval/?useMclip=true&query=%E9%BB%84%E8%89%B2%E3%81%84%E7%8C%AB&back=https%3A%2F%2Fknn5.laion.ai&index=laion5B

**Comment by dmarx** [comment]
@rom1504 @apolinario the m-clip release gave me a thought: maybe we could host mmc on pypi *with essentially none of the other perceptors installed at all*. Simple instructions for "finalizing" the mmc install could live in the README (as well as one-liners for specific perceptors PRN), and we could add a warning on import too.

My thinking here is if we ship the core tooling as a bare library, then anyone could attach the mocking utilities upstream to quickly make new perceptors drop-in-able if they aren't already, which conversely would make them trivial to add to mmc (since they'd already be hooked into a conformant API one way or another). 

Actually, it might be cleaner and simpler to isolate a simple mocking wrapper and package that for pypi?

I'm mostly just thinking out-loud now. Thoughts?

**Comment by apolinario** [comment]
I like the idea and spirit and I feel eventually if MMC gets way too many perceptors making some optional make a lot of sense. Now starting with all optional, I'm not sure - regardless I think your idea holds - just not sure if we ship empty or with some basics (OpenAI + OpenCLIP for e.g.) and let users further install from then on

**Comment by apolinario** [comment]
https://github.com/microsoft/UniCL

**Comment by apolinario** [comment]
https://github.com/goel-shashank/CyCLIP

**Comment by dmarx** [comment]
https://github.com/facebookresearch/omnivore

**Comment by dmarx** [comment]
https://github.com/microsoft/GLIP

**Comment by dmarx** [comment]
https://github.com/microsoft/RegionCLIP

**Comment by rom1504** [comment]
https://github.com/FacePerceiver/FaRL#use-farl-as-faceclip face clip

**Comment by dmarx** [comment]
https://github.com/Lednik7/CLIP-ONNX/tree/main/clip_onnx

**Comment by dmarx** [comment]
https://github.com/OFA-Sys/OFA

**Comment by apolinario** [comment]
Turkish CLIP https://github.com/yusufani/TrCLIP

**Comment by dmarx** [comment]
https://github.com/salesforce/LAVIS

**Comment by dmarx** [comment]
EVA-CLIP - https://github.com/baaivision/EVA/blob/master/clip/README.md

basically already api compliant
